### DISCUSSION TITLE
How to add foreign key on an array column of ids refering to another table

### DISCUSSION THREAD

**Discussion by foufrix** [supabase/supabase#18777] [Questions]
In one of my table, I have a column which is an array of Ids that I want to link to another table. Is it possible to do that or I am doing something wrong?

<img width="246" alt="image" src="https://github.com/supabase/supabase/assets/16646581/a515a3e2-0a43-44e3-bd5a-407435ed638d">

`skills[]` value are `[id1, id2, id3]` referring to another table with unique id value and other things.

Thanks in advance

**Comment by GaryAustin1** [comment]
It is not possible in Postgres.  The normal solution is to use a join table which has an fkey pointing to your original table and another one pointing to the table your array was referring to.  The create multiple row pairs for each id in the array.
